python/rmm/_lib/logger.pyx

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+
+from libcpp cimport bool
+
+
+cdef extern from "spdlog/common.h" namespace "spdlog::level" nogil:
+    cpdef enum logging_level "spdlog::level::level_enum":
+        """
+        The debug logging level for RMM.
```

Comment:
I chose the terminology deliberately, and it has been called this for years now. As the readme explains, RMM has two types of logging. Memory event logging is accomplished by using a `logging_resource_adaptor` (or by passing `logging=True` to `rmm.reinitialize()` in Python.

Debug logging is a lower-level diagnostic and debugging tool use by developers (possibly only me so far).

So we have to distinguish this type of logging somehow, I don't think we can just say "logging level".

Reply:
I forgot that we were already using this terminology. I still find it confusing that you can have debug logging at non-DEBUG levels, but not going to fight that battle in this PR since that's definitely out of scope here.